_posts/en/newsletters/2019-10-02-newsletter.md

```diff
@@ -0,0 +1,135 @@
+  co-author Gleb Naumenko.  Currently, Bitcoin nodes send each of their
+  peers the txids for all newly-seen transactions, resulting in each
+  node receiving many duplicate txid announcements.  This can be quite
+  bandwidth inefficient for Bitcoin's tens of thousands of nodes and
+  several hundred thousand transactions a day.  As [described
+  previously][erlay paper], minisketch-based set reconciliation allows
+  nodes to instead send a *sketch* of a set of txids that can be combined
+  with the txids the receiving peer already knows about to allow the
+  receiver to recover the txids it hasn't seen yet.  The size of the
+  sketch is roughly equal to the expected size of the txids that need to
+  be recovered, reducing txid-announcement bandwidth.  Erlay is a
+  proposal that suggests how to use this mechanism to achieve the best
+  mix of bandwidth efficiency and network robustness.  This draft BIP
+  describes the proposed implementation of the minisketch-based set
+  reconciliation between nodes, laying the foundation for the
+  implementation of erlay.  We encourage anyone with feedback to contact
+  the BIP authors either privately or on the mailing list.
+
+- **Full disclosure of fixed vulnerabilities affecting multiple LN
+  implementations:** several weeks ago, the developers of C-Lightning,
+  Eclair, and LND announced the previous discovery of an undisclosed
+  issue in each of their implementations, which they had each fixed in a
+  recent release.  At that time, they strongly encouraged their users to
+  upgrade (a message which Optech [relayed][optech ln warning]) and
+  promised a full disclosure in the future, which they have now done
+  with a [email][russell disclosure] by vulnerability discoverer and
+  C-Lightning developer Rusty Russell and a [blog post][lnd stay safe]
+  by LND developers Olaoluwa Osuntokun and Conner Fromknecht.
+
+    Briefly, the issue appears to have been that implementations did not
+    confirm that channel open transactions paid the correct script,
+    amount, or both.  Because of this, the implementations would accept
+    payments within the channel which they would later be unable to get
+    confirmed onchain, allow them to be defrauded.  As of this writing,
+    Optech is not aware of any reports that this issue was exploited
+    prior to the warning last month.  Now that the issue has
+    been disclosed, a [PR][BOLTS #676] has been opened to update the
+    specification to note that this check is needed.
+
+- **Optech taproot workshop:** last week, Optech held workshops in
+  both San Francisco and New York City teaching developers about the
+  schnorr signature scheme and other parts of the proposed
+  [taproot][bip-taproot] soft fork.  We thank Bryan Bishop for typing an
+  excellent [transcript][workshop transcript] of the NYC workshop.  We
+  are preparing videos and other educational material for release via a
+  blog post in the near future.
```

Comment:
Do we want to link to https://github.com/bitcoinops/taproot-workshop or save this for the upcoming blog post?

Reply:
I think save it. Let's release/announce all the material at the same time.